packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -78,3 +85,111 @@ class UnpackMacOS extends Target {
     }
   }
 }
+
+/// Tell cocoapods to re-fetch dependencies.
+class DebugMacOSPodInstall extends Target {
+  const DebugMacOSPodInstall();
+
+  @override
+  String get name => 'debug_macos_pod_install';
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.artifact(Artifact.flutterMacOSPodspec,
+      platform: TargetPlatform.darwin_x64,
+      mode: BuildMode.debug
+    ),
+    Source.pattern('{PROJECT_DIR}/macos/Podfile', optional: true),
+    Source.pattern('{PROJECT_DIR}/macos/Runner.xcodeproj/project.pbxproj'),
+    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/Flutter-Generated.xcconfig'),
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[
+    // No outputs because Cocoapods is fully responsible for tracking. plus there
```

Comment:
We'd need to right it such that the file contents changed as well. Just a stamp wont work since we're using file hashes, at that point it is probably close to the same as .flutter-plugins.

I've updated this with an explicit todo to add a planning phase so I can conditionally insert the pod rule. Then I can use the regular outputs.